SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -173,22 +162,18 @@ private bool FetchArgumentsAndRulesets(ProcessedArgs args, string configDir, ILo
 
                 throw;
             }
+            finally
+            {
+                Utilities.SafeDispose(server);
+            }
 
             return true;
         }
 
-        private static IDownloader GetDownloader(ProcessedArgs args)
-        {
-            string username = args.GetSetting(SonarProperties.SonarUserName, null);
-            string password = args.GetSetting(SonarProperties.SonarPassword, null);
-
-            return new WebClientDownloader(new WebClient(), username, password);
-        }
-
-        private void GenerateFxCopRuleset(SonarWebService ws, string projectKey, string requiredPluginKey, string language, string repository, string path, ILogger logger)
+        private void GenerateFxCopRuleset(ISonarQubeServer server, string projectKey, string requiredPluginKey, string language, string repository, string path, ILogger logger)
         {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The 'this' parameter (or 'Me' in Visual Basic) of 'TeamBuildPreProcessor.GenerateFxCopRuleset(ISonarQubeServer, string, string, string, string, string, ILogger)' is never used. Mark the member as static (or Shared in Visual Basic) or use 'this'/'Me' in the method body or at least one property accessor, if appropriate. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AMarkMembersAsStatic)